pipenv/cli/command.py

```diff
@@ -399,8 +400,8 @@ def shell(
 @pass_state
 def run(state, command, args):
     """Spawns a command installed into the virtualenv."""
-    from ..core import do_run
-
+    from ..core import do_run, warn_in_virtualenv
+    warn_in_virtualenv()
```

Comment:
This is not necessary, duplicate message is produced:
```
(.venv) λ C:\Users\frostming\.virtualenvs\pipenv-4vMlZWZY\Scripts\pipenv run pip list
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project. You can set PIPENV_IGNORE_VIRTUALENVS=1 to force pipenv to ignore that environment and create its own instead. You can set PIPENV_VERBOSITY=-1 to suppress this warning.
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project. You can set PIPENV_IGNORE_VIRTUALENVS=1 to force pipenv to ignore that environment and create its own instead. You can set PIPENV_VERBOSITY=-1 to suppress this warning.
Package    Version
---------- -------
Click      7.0
```

Reply:
Since putting it there, this is my experience:

```console
virtualenv /mnt/ramdisk/.venv; and source /mnt/ramdisk/.venv/bin/activate.fish; and pip install -e .; and pipenv install --dev
```

![image](https://user-images.githubusercontent.com/954310/53458872-bce6ca80-3a05-11e9-8d7b-45fbb72da579.png)

Reply:
I got this solved, see my patching https://github.com/pypa/pipenv/pull/3298/commits/3f053a355f0a6936939542ae791c8c6c2aec9424